src/provider/store/Actions.ts

```diff
@@ -1,43 +1,47 @@
-import {Action as ReduxAction} from 'redux';
-
 import {StoredNotification} from '../model/StoredNotification';
 import {Injector} from '../common/Injector';
 import {Inject} from '../common/Injectables';
 import {CollectionMap} from '../model/database/Database';
 import {SettingsMap} from '../model/StoredSetting';
 
 import {RootState} from './State';
-import {StoreAPI} from './Store';
-
-export const enum Action {
-    CREATE = '@@notifications/CREATE',
-    REMOVE = '@@notifications/REMOVE',
-    CLICK_NOTIFICATION = '@@notifications/CLICK_NOTIFICATION',
-    CLICK_BUTTON = '@@notifications/CLICK_BUTTON',
-    TOGGLE_VISIBILITY = '@@ui/TOGGLE_CENTER_WINDOW',
-}
+import {StoreAPI, AsyncAction, Action} from './Store';
 
-export class BaseAction<T extends Action> implements ReduxAction<Action> {
-    public readonly type: T;
```

Comment:
Something to note, by removing `type`, this may become harder to debug when minimised (since class names get mangled).

Maybe we're happy with that tradeoff, but should make sure that's an actual decision.

Reply:
The the base `Action` constructor takes a string to use as the type/name for debugging.

Reply:
^ this